cla_backend/apps/cla_auth/middleware.py

```diff
@@ -0,0 +1,39 @@
+import logging
+from json import dumps
+
+from rest_framework.response import Response
+from rest_framework import status
+from rest_framework.views import exception_handler
+from django.http import HttpResponse
+
+from mohawk.exc import TokenExpired
+
+log = logging.getLogger(__name__)
+
+
+class OBIEEHawkResponseMiddleware:
+    def process_exception(self, request, exception):
+        if isinstance(exception, TokenExpired):
+            resp = HttpResponse(dumps({'detail': 'invalid timestamp'}), 
+                status=401,
+                content_type='application/json')
+            resp['WWW-Authenticate'] = exception.www_authenticate
+            return resp
+        return None
+
+    def process_response(self, request, response):
+        is_hawk_request = request.META.get(
+            'HTTP_AUTHORIZATION', '').startswith('Hawk')
+        receiver = request.META.get('hawk.receiver', None)
+
+        if is_hawk_request and receiver:
```

Comment:
Can't happen - request.META['hawk.receiver'] is only set when the signature has been verified and a matching Client exists: https://github.com/ministryofjustice/cla_backend/blob/feature/obiee-csv-export-hawk/cla_backend/apps/cla_auth/auth.py#L44